src/app/modules/authorization/authorization.reducer.ts

```diff
@@ -7,35 +7,63 @@ export const authorizationFeatureKey = 'authorization';
 
 export interface State {
   authorizationToken?: string;
+  processing: {
+    signIn: boolean,
+    signUp: boolean,
+    signOut: boolean,
+  };
 }
 
 const authorizationTokenStorageKey = 'authorization';
 
 export const initialState: State = {
   authorizationToken: localStorage.getItem(authorizationTokenStorageKey) || undefined,
+  processing: {
+    signIn: false,
+    signUp: false,
+    signOut: false,
+  }
 };
 
 const authorizationReducer = createReducer(
   initialState,
-  on(
-    AuthorizationActions.signUpSuccess,
-    AuthorizationActions.signInSuccess,
+  on(AuthorizationActions.signInSuccess,
     (state, { authorizationToken }) => {
       localStorage.setItem(authorizationTokenStorageKey, authorizationToken);
-      return { ...state, authorizationToken };
+      return { ...state, authorizationToken, processing: { ...state.processing, signIn: false } };
+    }),
+  on(AuthorizationActions.signUpSuccess,
+    (state, { authorizationToken }) => {
+      localStorage.setItem(authorizationTokenStorageKey, authorizationToken);
+      return { ...state, authorizationToken, processing: { ...state.processing, signUp: false } };
+    }),
+  on(AuthorizationActions.signOutSuccess,
+    state => {
+      localStorage.removeItem(authorizationTokenStorageKey);
+      return { ...state, authorizationToken: undefined, processing: { ...state.processing, signOut: false } };
     }
   ),
-  on(
-    AuthorizationActions.signOutSuccess,
-    ErrorResponseActions.unauthorized,
+  on(AuthorizationActions.signIn, state => ({ ...state, processing: { ...state.processing, signIn: true } })),
+  on(AuthorizationActions.signUp, state => ({ ...state, processing: { ...state.processing, signUp: true } })),
+  on(AuthorizationActions.signOut, state => ({ ...state, processing: { ...state.processing, signOut: true } })),
+  on(AuthorizationActions.signInError,
+    state => ({ ...state, processing: { ...state.processing, signIn: false } })
+  ),
+  on(AuthorizationActions.signUpError,
+    state => ({ ...state, processing: { ...state.processing, signUp: false } })
+  ),
+  on(AuthorizationActions.signOutError,
+    state => ({ ...state, processing: { ...state.processing, signOut: false } })
+  ),
+  on(ErrorResponseActions.unauthorized,
     state => {
       localStorage.removeItem(authorizationTokenStorageKey);
       return { ...state, authorizationToken: undefined };
     }
   ),
 );
 
-export function reducer(state: State | undefined, action: Action) {
+export function reducer(state: State | undefined, action: Action): State {
   return authorizationReducer(state, action);
 }
 
```

Comment:
te flagi processing to sa flagi że np "trwa proces logowania"?

Reply:
Tak